Compiler/src/tfuncs.jl

```diff
@@ -450,6 +450,10 @@ end
                     return Const(true)
                 end
             end
+        # datatype_fieldcount is what `fieldcount` uses internally
+        # and returns nothing (!==0) for non-definite field counts.
+        elseif datatype_fieldcount(a1) === 0
```

Comment:
That comment seems to be false still, as it throws when given a NamedTuple:
```julia
julia> Base.datatype_fieldcount(NamedTuple{<:Any, Tuple}.body)
ERROR: ArgumentError: type does not have a definite number of fields
Stacktrace:
 [1] fieldcount
   @ ./runtime_internals.jl:1155 [inlined]
 [2] datatype_fieldcount(t::DataType)
   @ Base ./runtime_internals.jl:1119
 [3] top-level scope
   @ REPL[5]:1
```

Reply:
That looks like a bug in `datatype_fieldcount`; the intent seems to be for it not to throw.

Reply:
good catch. Fixed.